client/azure/azure.go

```diff
@@ -0,0 +1,132 @@
+package azure
```

Comment:
I'd like to make this flattened. `client.NewAzure` or `azure.NewClient` both look good to me. I don't see a reason to nest that package. We tend to do that way too much IMO.